Skip the checks if the news file was changed (#337)
twisted/towncrier: README.rst

    directory = "deprecation"
    name = "Deprecations"
    showcontent = true

Automatic pull request checks
-----------------------------

> To check if a feature branch is missing a news fragment, run::

Reviewer: ```suggestion
To check if a feature branch adds at least one news fragment, run::
```

Author: Good catch!

    towncrier check

By default this compares the current branch against ``origin/master``. You can use ``--compare-with`` if the trunk is named differently::

    towncrier check --compare-with origin/main

> The check succeeds if at least one of the following is true:
> 
> - The current branch adds at least one news fragment
> - The current branch changes the news file (e.g. the PR is a new release)

Reviewer: Maybe something like this.
My fear is that in the future we might modify the check and will forget to update the docs.
So instead of an exhaustive list of checks, maybe just mention that check will do nothing when NEWS is modified.

```suggestion
The check is automatically skipped when the main news file is modified inside the branch as this signals a release branch that is expected to not have news fragments.
```
